app.rb

```diff
@@ -0,0 +1,98 @@
+require_relative 'book'
+require_relative 'student'
+require_relative 'teacher'
+require_relative 'rental'
+require_relative 'classroom'
+
+class App
+  def initialize
+    @books = []
+    @people = []
+    @rentals = []
+  end
+
+  def list_all_books
+    @books.each do |book|
+      puts "Title: \"#{book.title}\", Author: #{book.author}"
+    end
+  end
+
+  def list_all_people
+    @people.each do |person|
+      puts "[#{person.type}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+    end
+  end
+
+  def create_a_person(choice, name, age)
+    if choice == '1'
+      person = create_a_student(name, age)
+      person.type = 'Student'
+    elsif choice == '2'
+      person = create_a_teacher(name, age)
+      person.type = 'Teacher'
+    else
+      puts 'wrong choice, choose from the option'
+    end
+    @people << person
+    puts 'Person created successfully'
+  end
+
+  def create_a_book(title, author)
+    book = Book.new(title, author)
+    @books << book
+    puts 'Book created successfully'
+  end
+
+  def create_a_rental
+    puts 'Select a book from the following list by number'
+    list_books
+    index_book = gets.chomp.to_i
+    puts 'Select a person from the following list by number (not id)'
+    list_people
+    index_person = gets.chomp.to_i
+    print 'Date: '
+    date = gets.chomp
+    rental = Rental.new(date, @books[index_book], @people[index_person])
+    @rentals << rental
+    puts 'Rental created successfully'
+  end
+
+  def list_all_rentals_for_a_given_person_id
+    puts 'ID of person: '
+    id = gets.chomp.to_i
+    puts 'Rentals: '
+    @rentals.each do |rental|
+      puts "Date: #{rental.date}, Book \"#{rental.book.title}\" by #{rental.book.author}" if rental.person.id == id
+    end
+  end
```

Comment:
Hi @unorjikingsley 👋🏼. I understand things might not be clear to you and you are in a hurry to move forward, but kindly note that you have to meet the requirements of this milestone to have it marked as completed😉.

<img width="1010" alt="image" src="https://github.com/unorjikingsley/School-Library/assets/64643995/6f58bc79-c4ab-4030-b0f2-565217330168">

It is a requirement that you pay attention to the single responsibility rule which inquires that a class should be assigned a single responsibility. This makes your codebase easily maintainable and usable. Currently, you have assigned the responsibility of input validation and requests to the main class, and what we are requesting is that you stay consistent with it by taking those responsibilities from your `app` and adding them to the main class😉.

You can do this by simply taking those actions that request and validates input from the `app` class to the `main` class where this responsibility has been assigned. You can then call these actions here in your app class where needed👍🏼.

Please do well to take a look at [this comment](https://github.com/unorjikingsley/School-Library/pull/6/files#r1280459754) left by the previous reviewer and make the changes where necessary👍🏼.